### PR TITLE
RevocationDB fixes

### PR DESCRIPTION
**Pull request by bryanchriswhite**

### CODE REVIEW

File: pkg/provider/identity.go [hunk 1]
```diff
@@ -82,21 +82,23 @@ type ServerConfig struct {
 type ServerOptions struct {
 	Config   ServerConfig
 	Ident    *FullIdentity
+	RevDB    *peertls.RevocationDB
```
Comment (egonelbre) [inline pkg/provider/identity.go]: This looks very weird to close "Options".

Comment (bryanchriswhite) [reply pkg/provider/identity.go]: do you have an alternate name or flow in mind?

File: pkg/provider/identity.go [hunk 2]
```diff
@@ -294,11 +296,14 @@ func (ic IdentityConfig) Run(ctx context.Context, interceptor grpc.UnaryServerIn
 	if err != nil {
 		return err
 	}
+	defer func() { _ = opts.RevDB.Close() }()
```
Comment (egonelbre) [inline pkg/provider/identity.go]: Check the error.

File: pkg/provider/identity.go (hunk 1 above)
Comment (egonelbre) [reply pkg/provider/identity.go]: not at the moment

File: pkg/provider/identity.go (hunk 2 above)
Comment (bryanchriswhite) [reply pkg/provider/identity.go]: What should we do with an error in a defer? Or should it not be in a defer?

Comment (egonelbre) [reply pkg/provider/identity.go]: There's no really nice way of handling them at the moment.

One way would be to use something like:

```
defer func() { err = utils.CombineErrors(err, opts.RevDB.Close()) }
```

The other would be to manually close everywhere.

Comment (bryanchriswhite) [reply pkg/provider/identity.go]: I don't understand how your defer example would do anything useful and I'm pretty sure we want to use a defer for this

Comment (egonelbre) [reply pkg/provider/identity.go]: Take a look at https://play.golang.org/p/0qh7EuHW7Ut. It relies on the behavior that you can change the returned `err` variable.

**Pull request merged**